subprojects/ide/src/crossVersionTest/groovy/org/gradle/plugins/ide/tooling/r47/ToolingApiIdeaModelCrossVersionSpec.groovy

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.plugins.ide.tooling.r47
+
+import org.gradle.integtests.tooling.fixture.TargetGradleVersion
+import org.gradle.integtests.tooling.fixture.ToolingApiSpecification
+import org.gradle.integtests.tooling.fixture.ToolingApiVersion
+import org.gradle.tooling.model.idea.IdeaContentRoot
+import org.gradle.tooling.model.idea.IdeaModule
+import org.gradle.tooling.model.idea.IdeaProject
+
+class ToolingApiIdeaModelCrossVersionSpec extends ToolingApiSpecification {
+
+    @ToolingApiVersion(">=4.7")
+    @TargetGradleVersion(">=4.7")
```

Comment:
It's not a blocking issue, but you can move these annotations to the class level. This way you can eliminate the duplication and you can still override the annotations if you specify different values for a specific method.